### PR TITLE
Variant report2

### PR DESCRIPTION
**Pull request by northwestwitch**
Improved code and variant report interface.

### CODE REVIEW

**Comment by northwestwitch** [comment]
I solved the testing issues. I think it was something related to me not having installed a library and not having removed some cached version of a test. Anyway, it works now! I've also improved the report thanks to comments from @henrikstranneheim .

**Comment by moonso** [comment]
Super! 🤸‍♀️

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@ All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
+- Added an analysis report page (html and PDF format) containing phenotype, gene panels and variants that are relevant to solve a case.
```
Comment (moonso) [inline CHANGELOG.md]: 👍

File: docs/user-guide/cases.md [hunk 2]
```diff
@@ -10,7 +10,10 @@ Your name will be tagged for each case that is assigned to you. For cases that a
 ----------
 
 ## Case
-This is the detailed view of one individual case. The intended use is to give a consise overview of the family under investigation and show recent activity related to the case.
+This is the detailed view of one individual case. The intended use is to give a concise overview of the family under investigation and show recent activity related to the case.
+
+## Case report page
+It can be visualized by clicking on the 'Visualize report' link on the left sidebar present on the case view. The case report contains a phenotype info panel as well as an overview of the gene panels used for the analysis. This page is showing also any variant users have been interacting with when working with the case: causative variants, variants pinned as candidates, ACMG or otherwise manually classified variants, as well as variants with any comment field. At the bottom of the report page the 'Dismissed variants' panel lists all variants that have been dismissed, with the reason(s) why they were rejected as candidates. By clicking on the top right button the case report can be exported and saved into a PDF file.
```
Comment (moonso) [inline docs/user-guide/cases.md]: Nice

File: scout/server/blueprints/cases/controllers.py [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 import query_phenomizer
 
 from scout.constants import (CASE_STATUSES, PHENOTYPE_GROUPS, COHORT_TAGS)
+from scout.constants.variant_tags import MANUAL_RANK_OPTIONS, DISMISS_VARIANT_OPTIONS, GENETIC_MODELS
 from scout.models.event import VERBS_MAP
 from scout.server.utils import institute_and_case
+from scout.server.blueprints.variants.controllers import variants_description, variants_filter_by_field
 
 STATUS_MAP = {'solved': 'bg-success', 'archived': 'bg-warning'}
 SEX_MAP = {'1': 'male', '2': 'female'}
```
Comment (moonso) [inline scout/server/blueprints/cases/controllers.py]: There is a SEX_MAP in [case_tags.py](https://github.com/Clinical-Genomics/scout/blob/master/scout/constants/case_tags.py).
please use that one 😺

File: scout/server/blueprints/cases/controllers.py [hunk 4]
```diff
@@ -93,6 +96,54 @@ def case(store, institute_obj, case_obj):
...
+    #get detailed info for the causatives:
+    causatives = variants_description(store, data['causatives'], case_obj, institute_obj)
+    data.update({'causatives_detailed': causatives})
+
+    #get detailed info for the pinned:
+    pinned = variants_description(store, data['suspects'], case_obj, institute_obj)
+    data.update({'pinned_detailed': pinned})
+
+    ## get variants for this case that are either classified, commented, tagged or dismissed.
+    all_variants = list(store.variants(case_id=case_obj['_id'], nr_of_variants=-1)) #all snv variants
```
Comment (moonso) [inline scout/server/blueprints/cases/controllers.py]: Let's add a new method to get the relevant variants directly

File: scout/server/blueprints/variants/controllers.py [hunk 5]
```diff
@@ -273,6 +273,57 @@ def variant(store, institute_obj, case_obj, variant_id):
     }
 
 
+def variants_filter_by_field(store, variants_list, field, case_obj = None, institute_obj= None):
+    """Given a list of variant objects return only those that have a key specified by "field" and a value of this field not empty."""
+
+    filtered_ids = []
+    for variant in variants_list:
+        if field in variant:
```
Comment (moonso) [inline scout/server/blueprints/variants/controllers.py]:
```python
if variant.get(field):
     do something 
```


Would work even if the value is nothing relevant

**Pull request merged**